generators/gae/index.js

```diff
@@ -141,8 +142,93 @@ module.exports = class extends BaseGenerator {
         return [_.kebabCase(this.baseName)];
     }
 
+    getMicroserviceFolders(input) {
+        if (!input) return null;
+        const destinationPath = this.destinationPath(input);
+        const files = shelljs.ls('-l', destinationPath);
+        const appsFolders = [];
+
+        files.forEach(file => {
+            if (file.isDirectory()) {
+                if (shelljs.test('-f', `${destinationPath}/${file.name}/.yo-rc.json`)) {
+                    try {
+                        const fileData = this.fs.readJSON(`${destinationPath}/${file.name}/.yo-rc.json`);
+                        if (fileData['generator-jhipster'].applicationType === 'microservice') {
+                            appsFolders.push(file.name.match(/([^/]*)\/*$/)[1]);
+                        }
+                    } catch (err) {
+                        this.log(chalk.red(`${file}: this .yo-rc.json can't be read`));
+                        this.debug('Error:', err);
+                    }
+                }
+            }
+        });
+
+        return appsFolders;
+    }
+
     get prompting() {
         return {
+            askForPath() {
+                if (this.abort) return;
+                if (this.applicationType !== 'gateway') return;
+                const done = this.async();
+                const messageAskForPath = 'Enter the root directory where the microservices are located';
+                const prompts = [
+                    {
+                        type: 'input',
+                        name: 'directoryPath',
+                        message: messageAskForPath,
+                        default: this.directoryPath || '../',
+                        validate: input => {
+                            const path = this.destinationPath(input);
+                            if (shelljs.test('-d', path)) {
+                                const appsFolders = this.getMicroserviceFolders(input);
+
+                                if (appsFolders.length === 0) {
+                                    return `No microservices are found in ${path}`;
+                                }
+                                return true;
+                            }
+                            return `${path} is not a directory or doesn't exist`;
+                        },
+                    },
+                ];
+
+                this.prompt(prompts).then(props => {
```

Comment:
```suggestion
                return this.prompt(prompts).then(props => {
```
And remove 
```
const done = this.async();
```
and 
```
done();
```

Returning the prompt promise is cleaner.

Reply:
@mashima : Not sure how this can be done it gives the lint error; `ESLint: Method 'askForPath' expected no return value.(consistent-return)` :thinking:

Reply:
Need to change the others returns too:
`if (this.abort) return undefined;`

Or add eslint disable rule.

Reply:
Ah, thanks. Done. :smile: